extensions/wikia/EmbeddableDiscussions/EmbeddableDiscussionsController.class.php

```diff
@@ -28,9 +28,67 @@ public static function onBeforePageDisplay() {
 		return true;
 	}
 
+	/**
+	 * Checks arguments for errors.
+	 * @param array $args
+	 * @oaram errorMessage Return parameter with the proper error message to show. Disregard if return is false
+	 * @return true if ok, false if error
+	 */
+	private static function checkArguments( array $args, &$errorMessage ) {
+		// mostrecent must be bool
+		if ( array_key_exists( 'mostrecent', $args ) && $args['mostrecent'] !== 'true' && $args['mostrecent'] !== 'false' ) {
+			$errorMessage = wfMessage( 'embeddable-discussions-parameter-error', 'mostrecent' )->plain() .
+				wfMessage( 'embeddable-discussions-parameter-error-boolean' )->plain();
+
+			return false;
+		}
+
+		// size must be integer in range
+		if ( array_key_exists( 'size', $args ) ) {
+			$size = $args['size'];
+
+			$errorMessage = wfMessage( 'embeddable-discussions-parameter-error', 'size' )->plain() .
+				wfMessage( 'embeddable-discussions-parameter-error-range',
+					self::ITEMS_MIN , self::ITEMS_MAX )->plain();
+
+			if ( !ctype_digit( $size ) ) {
+				return false;
+			} else if ( intval( $size ) > self::ITEMS_MAX || intval( $size ) < self::ITEMS_MIN ) {
+				return false;
+			}
+		}
+
+		// columns must be integer in range
+		if ( array_key_exists( 'columns', $args ) ) {
+			$columns = $args['columns'];
+
+			$errorMessage = wfMessage( 'embeddable-discussions-parameter-error', 'columns' )->plain() .
+				wfMessage( 'embeddable-discussions-parameter-error-range',
+					self::COLUMNS_MIN , self::COLUMNS_MAX )->plain();
+
+			if ( !ctype_digit( $columns ) ) {
+				return false;
+			} else if ( intval( $columns ) > self::COLUMNS_MAX || intval( $columns ) < self::COLUMNS_MIN ) {
+				return false;
+			}
+		}
+
+		return true;
+	}
+
 	public static function render( $input, array $args ) {
 		global $wgCityId;
 
+		$templateEngine = ( new Wikia\Template\MustacheEngine )->setPrefix( __DIR__ . '/templates' );
+
+		if ( !self::checkArguments( $args, $errorMessage ) ) {
+			return $templateEngine->clearData()
+				->setData( [
+					'errorMessage' => $errorMessage,
+				] )
+				->render( 'DiscussionError.mustache' );
+		}
+
 		$showLatest = !empty( $args['mostrecent'] ) && filter_var( $args['mostrecent'], FILTER_VALIDATE_BOOLEAN );
 		$itemCount = empty( $args['size'] ) ? self::ITEMS_DEFAULT : intval( $args['size'] );
 		$columns = empty( $args['columns'] ) ? self::COLUMNS_DEFAULT : intval( $args['columns'] );
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)